toolbox/include/base/WBBlock.h

```diff
@@ -0,0 +1,79 @@
+#ifndef WBT_WBIBLOCK_H
+#define WBT_WBIBLOCK_H
+
+#include "Block.h"
+#include <string>
+#include <memory>
+#include <iDynTree/Core/VectorFixSize.h>
+
+namespace wbt {
+    class WBBlock;
+    class Configuration;
+    class BlockInformation;
+    class RobotInterface;
+}
+
+namespace iDynTree {
+    class Twist;
+    class Transform;
+    class MatrixDynSize;
+    class VectorDynSize;
+}
+
+/**
+ * \struct iDynTreeRobotState WBBlock.h
+ *
+ * This struct contains the pointer to iDynTree objects used to configure the
+ * state of iDynTree::KinDynComputations objects.
+ */
+struct iDynTreeRobotState {
+    iDynTree::Twist*         m_baseVelocity   = nullptr;
```

Comment:
Do we really need these to be pointers?

I would also add a constructor (with dofs as parameter) and destructor to simplify memory management

Reply:
Removed pointers and switched to class objects·